Wikipedia/UI-V5/WMFRelatedTitleListDataSource.h

```diff
@@ -0,0 +1,37 @@
+//
+//  WMFRelatedTitleListDataSource.h
+//  Wikipedia
+//
+//  Created by Brian Gerstle on 9/4/15.
+//  Copyright (c) 2015 Wikimedia Foundation. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import <SSDataSources/SSArrayDataSource.h>
+#import "WMFArticleListDataSource.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class WMFRelatedSearchFetcher;
+@class MWKSavedPageList;
+@class MWKDataStore;
+@class MWKTitle;
+
+@interface WMFRelatedTitleListDataSource : SSArrayDataSource
+    <WMFArticleListDataSource>
+
+
+- (instancetype)initWithTitle:(MWKTitle*)title
+                    dataStore:(MWKDataStore*)dataStore
+                savedPageList:(MWKSavedPageList*)savedPageList
+    numberOfExtractCharacters:(NSUInteger)numberOfExtractCharacters;
+
+- (instancetype)initWithTitle:(MWKTitle*)title
+                    dataStore:(MWKDataStore*)dataStore
+                savedPageList:(MWKSavedPageList*)savedPageList
+    numberOfExtractCharacters:(NSUInteger)numberOfExtractCharacters
+                      fetcher:(WMFRelatedSearchFetcher*)fetcher NS_DESIGNATED_INITIALIZER;
```

Comment:
I guess I don't think there's anything dangerous about having the designated initializer publicly exposed.  The other option is to consolidate these into one method, and make the fetcher parameter optional (i.e. nullable), which is what I did in `WMFNearbyViewModel` for the location manager.  `WMFNearbyViewModel` however, has its own publicly exposed designated initializer.